### PR TITLE
Add php 8.1 Runtime

### PR DESCRIPTION
**Pull request by skywalkeretw**

### CODE REVIEW

**Comment by skywalkeretw** [comment]
@dgrove-oss Please review :)

File: README.md [hunk 1]
```diff
@@ -93,6 +104,10 @@ wskdev fresh -t local-php
 
 To use as docker action push to your own dockerhub account
 ```
+docker tag whisk/php8.0Action $user_prefix/action-php-v8.1
```
Comment (akrabat) [inline README.md:107]: `s/php8.0Action/php8.1Action/` ?

File: core/php8.1Action/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      pdo_pgsql \
+      soap \
+      zip \
+    \
+    && mkdir -p /usr/src/php/ext/mongodb \
+    && curl -fsSL https://pecl.php.net/get/mongodb-1.14.0 | tar xvz -C "/usr/src/php/ext/mongodb" --strip 1 \
+    && docker-php-ext-install -j$(nproc) mongodb \
+    \
+    && apt-get purge -y --auto-remove $PHPIZE_DEPS \
+    # && apt-get purge -y --auto-remove libclang-common-7-dev clang-7 llvm-7 llvm-7-dev \
```
Comment (akrabat) [inline core/php8.1Action/Dockerfile:79]: Can we delete this commented out line?

File: core/php8.1Action/runner.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    } catch (Throwable $e) {
+        file_put_contents('php://stderr', (string)$e);
+        $result = 'An error occurred running the action.';
+    }
+
+    // ensure that the sentinels will be on their own lines
+    file_put_contents('php://stderr', "\n");
+    file_put_contents('php://stdout', "\n");
+
+    // cast result to an object for json_encode to ensure that an empty array becomes "{}" & send to fd/3
```
Comment (akrabat) [inline core/php8.1Action/runner.php:89]: This comment is wrong as it doesn't do the cast here. Should be:

```
// send result to fd/3
```

File: core/php8.1Action/runner.php [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        }
+
+        // process the result
+        if (!is_array($result)) {
+            file_put_contents('php://stderr', 'Result must be an array but has type "'
+                . gettype($result) . '": ' . $result);
+            file_put_contents('php://stdout', 'The action did not return a dictionary or array.');
+            $result = (string)$result;
+        } else {
+            $result = json_encode((object)$result);
```
Comment (akrabat) [inline core/php8.1Action/runner.php:78]: Add comment before this line:

```
// cast result to an object for json_encode to ensure that an empty array becomes "{}
```

**Review by akrabat — CHANGES_REQUESTED** [review]
Looks great. Thank you for doing this. Just a few minor tweaks.

**Comment by skywalkeretw** [comment]
@akrabat I added your suggested fixes Thanks for the input

**Review by akrabat — APPROVED** [review]
LGTM 👍

**Review by ningyougang — APPROVED** [review]

**Comment by skywalkeretw** [comment]
Would someone be so kind and merge the pull request thx :)

**Pull request merged**